cli/commands/genesis/payload.go

```diff
@@ -69,137 +67,107 @@ func AddExecutionPayload(chainSpec ChainSpec, elGenesisPath string, config *cmtc
 	if err = ethGenesis.UnmarshalJSON(genesisBz); err != nil {
 		return errors.Wrap(err, "failed to unmarshal eth1 genesis")
 	}
 	genesisBlock := ethGenesis.ToBlock()
 
-	// Create the execution payload.
+	// Create the executable data from the EL genesis file.
 	payload := gethprimitives.BlockToExecutableData(
 		genesisBlock,
 		nil,
 		nil,
 		nil,
 	).ExecutionPayload
 
-	appGenesis, err := genutiltypes.AppGenesisFromFile(
-		config.GenesisFile(),
-	)
+	appGenesis, err := genutiltypes.AppGenesisFromFile(config.GenesisFile())
 	if err != nil {
 		return errors.Wrap(err, "failed to read genesis doc from file")
 	}
 
-	// create the app state
-	appGenesisState, err := genutiltypes.GenesisStateFromAppGenesis(
-		appGenesis,
-	)
+	// Create the app state.
+	appGenesisState, err := genutiltypes.GenesisStateFromAppGenesis(appGenesis)
 	if err != nil {
 		return err
 	}
 
 	genesisInfo := &types.Genesis{}
-
-	if err = json.Unmarshal(
-		appGenesisState["beacon"], genesisInfo,
-	); err != nil {
+	if err = json.Unmarshal(appGenesisState["beacon"], genesisInfo); err != nil {
 		return errors.Wrap(err, "failed to unmarshal beacon state")
 	}
-	// Inject the execution payload.
-	eph, err := executableDataToExecutionPayloadHeader(
-		chainSpec.GenesisForkVersion(),
-		payload,
-		chainSpec.MaxWithdrawalsPerPayload(),
-	)
+
+	// Inject the execution payload from the executable data.
+	eph, err := executableDataToExecutionPayloadHeader(payload)
 	if err != nil {
 		return errors.Wrap(err, "failed to convert executable data to execution payload header")
 	}
-	if eph == nil {
-		return errors.New("failed to get execution payload header")
-	}
+	// if eph == nil {
+	// 	return errors.New("failed to get execution payload header")
+	// }
 	genesisInfo.ExecutionPayloadHeader = eph
 
-	appGenesisState["beacon"], err = json.Marshal(genesisInfo)
-	if err != nil {
+	if appGenesisState["beacon"], err = json.Marshal(genesisInfo); err != nil {
 		return errors.Wrap(err, "failed to marshal beacon state")
 	}
 
-	if appGenesis.AppState, err = json.MarshalIndent(
-		appGenesisState, "", "  ",
-	); err != nil {
+	if appGenesis.AppState, err = json.MarshalIndent(appGenesisState, "", "  "); err != nil {
 		return err
 	}
 
 	return genutil.ExportGenesisFile(appGenesis, config.GenesisFile())
 }
 
-// Converts the eth executable data type to the beacon execution payload header
-// interface.
+// Converts the eth executable data type to the beacon execution payload header.
 func executableDataToExecutionPayloadHeader(
-	forkVersion common.Version,
 	data *gethprimitives.ExecutableData,
-	// todo: re-enable when codec supports.
-	_ uint64,
 ) (*types.ExecutionPayloadHeader, error) {
-	eph := &types.ExecutionPayloadHeader{}
-
-	// We do not support fork versions before Deneb and after Electra.
-	if version.IsAfter(forkVersion, version.Electra()) ||
-		version.IsBefore(forkVersion, version.Deneb()) {
-		return nil, types.ErrForkVersionNotSupported
```

Comment:
noting that as neither the geth `ExecutableData` nor the bkit `ExecutionPayloadHeader` objects are versioned, it doesn't make sense for the CLI command to be opinionated to versions